fuse/utils/multiprocessing/linux_shared_memory.py

```diff
@@ -0,0 +1,92 @@
+import os
+from os.path import join, basename
+import shutil
+
+# from multiprocessing import Lock
+from filelock import FileLock
+from glob import glob
+import psutil
+
+SHARED_MEM_FILES_PREFIX = "OUR_SHARED_MEM_@"
+SHARED_MEM_ACTIVE_ENV_VAR = "ACTIVATE_OUR_SHARED_MEM"
+SHM_BASE_DIR = "/dev/shm/"
+
+G_lock = FileLock(join(SHM_BASE_DIR, "our_shared_mem_file_lock"))  # Lock()
+
+
+def get_shared_mem_file_path(file_path: str) -> str:
+    """
+    copies the file (if needed) to /dev/shm/[filename] which effectively make linux os to load it into RAM
+    every following access to the file will actually be on RAM which is much faster
+
+    note - it's up to us to clean up the files, otherwise they will continue to consume RAM until a reboot!
+
+    a nice guide about this topic: https://datawookie.dev/blog/2021/11/shared-memory-docker/
+
+    """
+
+    if SHARED_MEM_ACTIVE_ENV_VAR not in os.environ:
+        return file_path
+
+    if os.environ[SHARED_MEM_ACTIVE_ENV_VAR].lower() not in ["1", "t", "true"]:
+        return file_path
+
+    with G_lock:
+        assert os.path.isfile(
+            file_path
+        ), f"file_path does not point to a file: file_path={file_path}"
+
+        src_file_size_bytes = os.stat(file_path).st_size
+        dest = join(SHM_BASE_DIR, SHARED_MEM_FILES_PREFIX + basename(file_path))
```

Comment:
I wonder if we want to keep the entire path structure (to avoid conflicts and keep the same relative path). 
For example:
file /mnt/bmfmbio/ab_design/1.fasta
Will be copied to /dev/shm/$SHARED_MEM_FILES_PREFIX/mnt/bmfmbio/ab_design/1.fasta

Reply:
was considering that - I did remember checking that /dev/shm does support directories in it

was a bit reluctant to do it because I run "remove" commands which delete files, so didn't want '..' paths to somehow escape it

Reply:
trying it now

Reply:
done - now full path will be stored under the shared memory directory.